### ISSUE TITLE
baseSepolia not compatible with Type "Chain"

### ISSUE THREAD

**Issue by humblemodulo** [wevm/viem#2474]
Works fine for me: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzjAnmApnAwgCwIbAB2cAvnAGZQQhwDkAbsOiLQFCiSyJwBGeAzugDK6SABtgeUhSo0GTEAHoAxviL82rZREL94qgrrgBeOAG0+gkeMkBdOALjbd8fnhjB+5JvzhR0eAAmOmIo2GqEZrasrEA

Can you please fork that playground and break it? Try double check your tsconfig too.

_Originally posted by @jxom in https://github.com/wevm/viem/issues/2127#issuecomment-2054650693_
            

Using `"viem": "^2.17.0",` and it's definitely not happy.  Any adjustment to be made?

Granted it's "only" a linting issue, but still.  The full error is as follows:

```
Type error: Type 'readonly [{ blockExplorers: { readonly default: { readonly name: "Basescan"; readonly url: "https://sepolia.basescan.org"; readonly apiUrl: "https://api-sepolia.basescan.org/api"; }; }; contracts: { ...; }; ... 10 more ...; readonly network: "base-sepolia"; }]' is not assignable to type 'readonly [Chain, ...Chain[]]'.
  Type at position 0 in source is not compatible with type at position 0 in target.
    Type '{ blockExplorers: { readonly default: { readonly name: "Basescan"; readonly url: "https://sepolia.basescan.org"; readonly apiUrl: "https://api-sepolia.basescan.org/api"; }; }; contracts: { ...; }; ... 10 more ...; readonly network: "base-sepolia"; }' is not assignable to type 'Chain'.
```

**Comment by jxom** [comment]
What is the issue? Please use the issue template and provide a minimal reproduction.

**Comment by humblemodulo** [comment]
@jxom apologies.  It's exactly the same as the referenced quote post.  Do you still need an example?

**Comment by jxom** [comment]
Yes. Break this playground: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzjAnmApnAwgCwIbAB2cAvnAGZQQhwDkAbsOiLQFCiSyJwBGeAzugDK6SABtgeUhSo0GTEAHoAxviL82rZREL94qgrrgBeOAG0+gkeMkBdOALjbd8fnhjB+5JvzhR0eAAmOmIo2GqEZrasrEA

**Comment by humblemodulo** [comment]
Well, that's frustrating.  I'm unable to "break" the TS example.  I've tried copying the `tsconfig` to mine and vice versa.  Nothing breaks it.

The only thing I can say verifiably is that most other `Chain`s I've tried (e.g. mainnet, arbitrum, bsc, polygon) result in no TS warnings locally (VS Code, Cursor), but `baseSepolia` (and now `optimism`) results in the above posted warning, and fails linting during the build process.

I can circumvent it with `chains: [baseSepolia as any as Chain]` but I'm more curious as to why it's a problem in the first place. 🤔
